### PR TITLE
Fixing log_out problems on mac os devices

### PR DESCRIPTION
**Pull request by madcat1991**
There is a problem with log_out function on mac os devices. The problem was discussed here:  https://github.com/veusz/veusz/issues/54 . In this commit I just added a check for OSError

### CODE REVIEW

**Comment by madcat1991** [comment]
I will check test the code tomorrow. This is just to show that we can leave everything as it was (except the deletion of the session file)

File: telethon/telegram_client.py [hunk 1]
```diff
@@ -315,22 +316,27 @@ def sign_up(self, phone_number, code, first_name, last_name=''):
...
         try:
             self(LogOutRequest())
             self.disconnect()
-            if not self.session.delete():
-                return False
-
-            self.session = None
-            return True
+        except OSError as e:
+            # macos issue: https://github.com/veusz/veusz/issues/54
```
Comment (Lonami) [inline telethon/telegram_client.py]: For some consistency change the comments to:
```python
            # macOS issue: https://github.com/veusz/veusz/issues/54
            # Socket has been already closed (Errno 57)
            # Fail on any other error
```

File: telethon/telegram_client.py [hunk 2]
```diff
@@ -315,22 +316,27 @@ def sign_up(self, phone_number, code, first_name, last_name=''):
...
+            # socket has been already closed (Errno 57)
+            # if any other - fail
+            if e.errno != errno.ENOTCONN:
+                raise e
```
Comment (Lonami) [inline telethon/telegram_client.py]: You can have simply:
```python
                raise
```

**Comment by Lonami** [comment]
Seems to work okay under Android so it should work on Linux/Windows too. I'll merge once you confirm it works on macOS.

**Comment by madcat1991** [comment]
Tested on macOS, everything is ok. Also updated the comment

File: telethon/telegram_client.py (hunk 2 above)
Comment (madcat1991) [inline telethon/telegram_client.py]: Done!

File: telethon/telegram_client.py (hunk 1 above)
Comment (madcat1991) [inline telethon/telegram_client.py]: Done!

**Comment by Lonami** [comment]
Just letting you know I moved the `e.errno != errno.ENOTCONN` check to the most basic levels to make sure things don't go wrong when it crashes.

**Comment by madcat1991** [comment]
Ok, thanks

**Pull request merged**